demo/GraphTutorial/app/src/main/java/com/example/graphtutorial/GraphHelper.java

```diff
@@ -90,34 +72,49 @@ public void getCalendarView(String accessToken,
         final List<Option> pagingOptions = new LinkedList<Option>();
         pagingOptions.add(new HeaderOption("Prefer", "outlook.timezone=\"" + timeZone + "\""));
 
-        final ICallback<IEventCollectionPage> pagingCallback = new ICallback<IEventCollectionPage>() {
-            @Override
-            public void success(IEventCollectionPage eventCollectionPage) {
-                allEvents.addAll(eventCollectionPage.getCurrentPage());
-
-                IEventCollectionRequestBuilder nextPage =
-                        eventCollectionPage.getNextPage();
-
-                if (nextPage == null) {
-                    callback.success(allEvents);
-                }
-                else{
-                    nextPage.buildRequest(pagingOptions)
-                            .get(this);
-                }
-            }
-
-            @Override
-            public void failure(ClientException ex) {
-                callback.failure(ex);
-            }
-        };
+        final CompletableFuture<List<Event>> future = new CompletableFuture<>();
```

Comment:
you don't need this additional future as the SDK already returns a future, just return the future the SDK returns

Reply:
Ok have a look at my refactor.